gslib/tests/test_shim_util.py

```diff
@@ -1155,12 +1156,41 @@ def test_decryption_key_gets_converted_to_flag(self):
       flags, _ = self._fake_command._translate_boto_config()
       self.assertEqual(flags, ['--decryption-keys=key1,key12,key100'])
 
-  def test_boto_config_translation_for_supported_fields(self):
+  @mock.patch.object(boto_util, 'UsingGsHmac')
+  def test_gs_hmac_auth_env_set_correctly(self, mock_using_gs_hmac):
     with _mock_boto_config({
         'Credentials': {
-            'aws_access_key_id': 'AWS_ACCESS_KEY_ID_value',
-            'aws_secret_access_key': 'AWS_SECRET_ACCESS_KEY_value',
-            'use_client_certificate': True,
+            'gs_access_key_id': 'unmapped',
+            'gs_secret_access_key': 'mapped',
+        }
+    }):
+      # Another use case that would have been good for parameterized.
+      # UsingGsHmac ultimately controls whether the gs HMAC creds
+      # are mapped, so the first key, gs_access_key_id, will not be
+      # mapped, while the second, gs_secret_access_key will.
+      mock_using_gs_hmac.side_effect = [False, True]
```

Comment:
I think this would be better as two separate tests, or maybe even just one -- it looks like the True case is covered below, so we could probably just check the False case here.

Reply:
Just made this a single test for the negative case.